scripts/dangerjs/checkChangelog.ts

```diff
@@ -26,32 +27,42 @@ const getMalformedChangelogEntries = async (danger): Promise<string[]> => {
   return addedLines.map(line => line.content).filter(content => content.startsWith('+-') && !validEntry.test(content));
 };
 
-const getAddedLinesFromChangelog = async (danger): Promise<{ content: string; ln: number }[]> => {
+const getAddedLinesFromChangelog = async (danger: DangerDSLType): Promise<{ content: string; ln: number }[]> => {
   return danger.git.structuredDiffForFile(CHANGELOG_FILE).then(changelogDiff => {
-    return changelogDiff.chunks.reduce((acc, chunk) => {
-      const filteredLines = chunk.changes.filter(change => change.type === 'add');
-      return acc.concat(filteredLines);
-    }, []);
+    if (changelogDiff) {
+      return changelogDiff.chunks.reduce((acc, chunk) => {
+        const filteredLines = chunk.changes.filter(change => change.type === 'add');
+        return acc.concat(filteredLines);
+      }, []);
+    }
+
+    return [];
   });
 };
 
 export default async ({ danger, fail, warn }: DangerJS) => {
-  // Check for a CHANGELOG entry
-  const hasChangelog = danger.git.modified_files.some(f => f === CHANGELOG_FILE);
+  // Check for a CHANGELOG entry for changes inside /packages/fluentui
+  const changes = [...danger.git.created_files, ...danger.git.deleted_files, ...danger.git.modified_files].filter(
```

Comment:
1. Does danger always use UNIX style paths?
2. Deleting the changelog file should be an error. Also, checking the created files only seems like it would be useful for this specific PR, since after that the file will already exist.

Reply:
1. danger uses git directly which normalizes to UNIX style paths

Reply:
2a. deleting changelog SHOULD be error - but i'm not really doing much to the logic here
2b. creating files can occur post merge of repo - we'll want to check for these changes

Reply:
For 2 I misunderstood what it was doing, sorry